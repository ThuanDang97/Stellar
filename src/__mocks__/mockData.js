export const TOOLTIP_LOREM =
  'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'

export const SELECT_OPTIONS = [
  { value: '1', name: 'Pay monthly' },
  { value: '3', name: 'Pay 3 monthly' },
  { value: '6', name: 'Pay 6 monthly' },
  { value: '12', name: 'Pay annually' },
]

export const USER_FULL_NAME = 'John Wick'

export const AVATAR_DEFAULT = {
  firstName: 'John',
  lastName: 'Wick',
}

export const AVATAR_IMAGE = {
  imageUrl: 'https://picsum.photos/20',
}

export const NETFLIX_REPORT = {
  name: 'netflix',
  period: 'report monthly',
}

export const GOAL_ITEM = {
  title: 'Improve My Credit Score',
  iconUrl: '/icons/piggy-bank',
}

export const BANK_NUMBER = '123456789111'

export const BANK_NAME = 'Chase'

export const INPUT_VALUES = {
  email: 'test@gmail.com',
  text: 'example',
  password: 'example1password',
  image:
    'data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e',
}

export const MEDIA_IMAGES = {
  netflix: '/images/netflix.png',
  hulu: '/images/hulu.png',
}

export const ADD_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Holo',
    amount: 77,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const MY_GOALS = [
  {
    id: '1',
    goal: 'Buy a car',
    scoreNeeded: 703,
    pointNeeded: 'READY',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    goal: 'Buy a house',
    scoreNeeded: 720,
    pointNeeded: '23 PTS AWAY',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 3434543645,
  },
]

export const LINKED_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    draftDate: '1st',
    status: 'Unpaid',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Hulu',
    amount: 32,
    draftDate: '4th',
    status: 'Paid',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const EXITS_EMAIL = 'thanh@endurance.com'

export const NUMBER_BANK = [
  {
    id: '1',
    bankNumber: '12435324234',
    bankName: 'Chase',
  },
]

export const ACCOUNT_LOGIN = [
  {
    userId: '1',
    email: 'khanhtam304n@gmail.com',
    password: 'Khanhtam@123',
  },
  {
    userId: '2',
    email: 'khanhtam3041n@gmail.com',
    password: 'Khanhtam@123',
  },
]

export const USER = {
  userId: '4',
  userName: 'Khanh Tam',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

export const GOAL = ['g1', 'g2']

export const MARKETING = ['m1', 'm2']

export const USER_ID = '123'

export const DATE_TIMER = new Date('08/28/2022')

export const USER_GOAL = { userId: USER_ID, goals: GOAL }

export const USER_MARKETING = { userId: USER_ID, goals: MARKETING }

export const ADDRESS = {
  address: '123 street',
  phone: '1231231231',
}

export const SSN_NUMBER = '123123123'

export const SUBSCRIPTION_PLAN = 'flex'

export const USER_SUBSCRIPTION_PLAN = {
  userId: USER_ID,
  subscriptionPlan: SUBSCRIPTION_PLAN,
}

export const CHART_SCORE = {
  data: [
    { month: 'M', score: 720 },
    { month: 'A', score: 740 },
    { month: 'M', score: 768 },
    { month: 'J', score: 785 },
    { month: 'J', score: 810 },
    { month: 'A', score: 830 },
  ],
}

export const LINK_TOKEN = 'sand-box-123'

export const BANK_ID = 'id_45'

export const BANK_INFO = {
  id: BANK_ID,
  bankNumber: BANK_NUMBER,
  bankName: BANK_NAME,
}
export const FICO_DATE_UPDATE = 'Last Updated Aug, 12, 2021'

export const NOTIFICATIONS = [
  {
    id: '1',
    title: 'Credit Score Update',
    desc: 'Your credit score increased by...',
    dateTime: '1 hours',
    isCheck: false,
  },
  {
    id: '2',
    title: 'Bill Pay',
    desc: 'We noticed you missed a payment...',
    dateTime: '3 days ago',
    isCheck: false,
  },
  {
    id: '3',
    title: 'Offer',
    desc: 'Look like you may quality for...',
    dateTime: '10 days ago',
    isCheck: false,
  },
  {
    id: '4',
    title: 'Offer',
    desc: 'Look like you may quality for offer',
    dateTime: '11 days ago',
    isCheck: false,
  },
]

export const APPLICATION_INFO = {
  title: 'Netflix',
  image: '/images/netflix.png',
}

export const APPLICATION_CONNECT_ACCOUNT = {
  userId: '1',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

// Min score: 300 - Max score 850
export const FICO_SCORE = 500

export const IDENTITY = {
  firstName: 'John',
  lastName: 'Wick',
  dateOfBirth: '02/02/2000',
  ssn: 123123123,
}

export const SSN_USER = {
  userId: USER_ID,
  ssn: SSN_NUMBER,
}

export const USER_IDENTITY = {
  userId: USER_ID,
  identity: IDENTITY,
}

export const ADDRESS_USER = {
  address: '604 Nui Thanh',
  phoneNumber: '+10949328492',
}

export const FICO_UNLOCK_SCORE = 708

export const LIST_BANKS_NUMBER = [
  { id: 'Chase', bankNumber: '423432432423232' },
  { id: 'Bank of America', bankNumber: '111122223333000' },
  { id: 'CitiBank Online', bankNumber: '111122223333111' },
]

export const INFO_PAYMENTS = {
  id: '123',
  money: '15.99',
  bankName: 'Chase',
  payDate: new Date(),
  numberPaymentsMonth: '3',
  appInfo: {
    title: 'Netflix',
    image: '/images/netflix.png',
  },
  status: 'UNPAID',
}

export const BILL_INFO = {
  title: 'Netflix',
  amount: 10,
}

export const ADDED_BILL_ID = '1234'

export const USER_DETAIL = {
  id: '123',
  amount: 10,
  draftDate: 31,
  bankNumber: BANK_NUMBER,
}

export const EDIT_LINK_BILL = {
  id: '123',
  amount: 10,
}
