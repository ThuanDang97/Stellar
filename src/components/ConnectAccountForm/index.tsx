import React, { memo, FC, useMemo } from 'react'
import { useTheme } from 'styled-components'

// Components
import Button from '@components/Button'
import Input from '@components/Input'
import ErrorMessage from '@components/ErrorMessage'
import { BoxShadowStyled, ViewStyled } from '@components/styled-components'
import Avatar from '@components/Avatar'
import { TitlePopupStyled } from '@components/Title/TitleStyled'

// Props type
import { ConnectAccountProps } from '@self-types/components/ConnectAccount.props'

const ConnectAccountForm: FC<ConnectAccountProps> = ({
  email,
  password,
  errorMessage,
  nameApplication,
  imageApplication,
  isShadowType,
  isEditModal,
  handleChangeEmail,
  handleChangePassword,
  handleSubmit,
  onCloseConnectAccountForm,
}) => {
  const theme = useTheme()

  // Render title connect account form
  const renderTitleForm = useMemo(() => {
    return (
      <ViewStyled
        display="flex"
        justifyContent="center"
        flexDirection="column"
        pLeft={50}
        pRight={50}
      >
        <Avatar imageUrl={imageApplication} width={80} height={80} />
        <TitlePopupStyled
          fontSize={theme.typography.fontSize.md}
          color={theme.colors.black}
          lineHeight={theme.typography.lineHeight.md}
          fontWeight={theme.typography.fontWeight.bold}
          mBottom={theme.metrics.dimensions.lg}
        >
          {`${nameApplication} Account`}
        </TitlePopupStyled>
      </ViewStyled>
    )
  }, [
    imageApplication,
    nameApplication,
    theme.colors.black,
    theme.metrics.dimensions.lg,
    theme.typography.fontSize.md,
    theme.typography.fontWeight.bold,
    theme.typography.lineHeight.md,
  ])

  return (
    <BoxShadowStyled
      data-testid="connectAccountForm"
      as="form"
      onSubmit={handleSubmit}
      noValidate
      shadowType={isShadowType ? 'regular' : 'none'}
      padding={isEditModal ? 0 : theme.metrics.dimensions.lg}
      textAlign="center"
      mLeft={theme.metrics.dimensions.md}
      mRight={theme.metrics.dimensions.md}
      minWidth={theme.metrics.width.xxl}
      maxWidth={400}
      bgColor={theme.colors.white}
    >
      {renderTitleForm}
      <Input
        placeholder="Email Address"
        type="email"
        name="email"
        iconUrl="/icons/envelope"
        value={email}
        onChange={handleChangeEmail}
        error={errorMessage?.email}
        mTop={theme.metrics.dimensions.xs}
      />
      <Input
        placeholder="Password"
        mTop={theme.metrics.dimensions.lg}
        isLogin
        type="password"
        name="password"
        iconUrl="/icons/icon-lock"
        value={password}
        onChange={handleChangePassword}
        error={errorMessage?.password}
      />
      {(errorMessage?.email || errorMessage?.password) && (
        <ErrorMessage error={errorMessage?.email || errorMessage?.password} />
      )}

      {isEditModal ? (
        <ViewStyled
          width={theme.metrics.width.full}
          display="flex"
          justifyContent="flex-end"
          gap={theme.metrics.dimensions.base}
          mTop={theme.metrics.dimensions.xxl}
          mBottom={theme.metrics.dimensions.md}
        >
          <Button
            bgColorHover={theme.colors.ironLight}
            onClick={onCloseConnectAccountForm}
            title="Cancel"
            color={theme.colors.shark}
            variant="dark"
          />
          <Button
            type="submit"
            title="Connect Account"
            disabled={!(email && password)}
          />
        </ViewStyled>
      ) : (
        <Button
          mTop={theme.metrics.dimensions.xxl}
          {...(email && { cursor: 'pointer' })}
          type="submit"
          title={`Connect ${nameApplication} Account`}
          disabled={!(email && password)}
          btnWidth={theme.metrics.width.full}
        />
      )}
    </BoxShadowStyled>
  )
}

export default memo(ConnectAccountForm)
