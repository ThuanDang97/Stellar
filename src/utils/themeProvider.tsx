import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
