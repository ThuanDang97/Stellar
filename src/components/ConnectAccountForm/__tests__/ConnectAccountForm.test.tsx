import { cleanup, fireEvent, screen } from '@testing-library/react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import ConnectAccountForm from '@components/ConnectAccountForm'

// Mocks
import { INPUT_VALUES } from '@mocks/mockData'

describe('ConnectAccountForm Component', () => {
  const props = {
    handleChangeEmail: jest.fn(),
    handleChangePassword: jest.fn(),
    handleSubmit: jest.fn(),
    email: '',
    password: '',
  }

  beforeEach(() => {
    renderWithTheme(<ConnectAccountForm {...props} />)
  })

  afterEach(() => {
    cleanup()
  })

  it('should render input email of application', () => {
    const emailInput = screen.getAllByTestId('iconInput')

    expect(emailInput[0]).toBeInTheDocument()
  })

  it('should call change input email of application', () => {
    const input = screen.getAllByTestId('iconInput')

    fireEvent.change(input[0], {
      target: { value: INPUT_VALUES.email },
    })

    expect(props.handleChangeEmail).toBeCalled()
  })

  it('should render input password of application', () => {
    const passwordInput = screen.getAllByTestId('iconInput')
    expect(passwordInput[1]).toBeInTheDocument()
  })

  it('should call change input password of application', () => {
    const input = screen.getAllByTestId('iconInput')

    fireEvent.change(input[1], {
      target: { value: INPUT_VALUES.email },
    })

    expect(props.handleChangePassword).toBeCalled()
  })

  it('should call submit form', () => {
    const button = screen.getByTestId('button')

    fireEvent.submit(button)

    expect(props.handleSubmit).toBeCalled()
  })

  it('should render button submit', () => {
    const button = screen.getByTestId('button')

    expect(button).toBeInTheDocument()
  })

  it('should render ErrorMessage', () => {
    const newProps = {
      ...props,
      errorMessage: { email: 'Invalid', password: 'Invalid' },
      email: '',
      password: '',
    }

    renderWithTheme(<ConnectAccountForm {...newProps} />)

    const ErrorMessage = screen.getByTestId('errorMessage')

    expect(ErrorMessage).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = renderWithTheme(<ConnectAccountForm {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
