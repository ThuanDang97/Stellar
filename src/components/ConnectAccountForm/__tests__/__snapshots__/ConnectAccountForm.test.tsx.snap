// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectAccountForm Component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 fTVJlI HQsEK"
    data-testid="connectAccountForm"
    novalidate=""
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 cafaom"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 gjRQva"
        cursor="pointer"
        height="80"
        width="80"
      >
        <span
          class="AvatarStyled-sc-1ppokaq-1 bOiqfJ"
          data-testid="avatarDefault"
        />
      </div>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 TitleStyled__TitlePopupStyled-sc-t58pjn-2 XkRb kyLpEc iyHIHp"
        color="#000"
        font-size="20"
        font-weight="700"
      >
        undefined Account
      </span>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 gJtdND brcjID"
      color="#203F59"
      data-testid="inputIcon"
      display="block"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <input
        class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
        data-testid="iconInput"
        name="email"
        placeholder="Email Address"
        type="email"
        value=""
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 kdxmlR brcjID"
      color="#203F59"
      data-testid="inputIcon"
      display="block"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <input
        class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
        data-testid="iconInput"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      class="ButtonStyled-sc-1ys9v52-0 gmKOVy"
      data-testid="button"
      disabled=""
      type="submit"
    >
      Connect undefined Account
    </button>
  </form>
</DocumentFragment>
`;
